Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -179,13 +229,28 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
                 for (int iy=0; iy<=depos_order; iy++){
                     for (int ix=0; ix<=depos_order; ix++){
                         amrex::Gpu::Atomic::AddNoRet(
-                            &rho_arr(lo.x+i+ix, lo.y+j+iy, lo.z+k+iz),
+                            &rho_tmp(lo.x+i+ix, lo.y+j+iy, lo.z+k+iz),
                             sx[ix]*sy[iy]*sz[iz]*wq);
                     }
                 }
             }
 #endif
         }
+
+#if defined(AMREX_USE_GPU)
+        if (shared_mem_bytes > 0) {
+          __syncthreads();
+
+          volatile amrex::Real *vs = shared;
+          for (int i = threadIdx.x; i < rho_box.numPts(); i += blockDim.x) {
+            amrex::Real r = vs[i];
+            // filter out extraneous atomic operations
+            if (r != 0.0)
```

Comment:
This turns out not to be an optimization. I suspect that's because it prevents coalesced memory access that can do the atomic over entire wide vectors, rather than individual elements. I'm removing it.